console/src/main/scripts/plugins/metrics/ts/services/metricsService.ts

```diff
@@ -91,6 +91,18 @@ module HawkularMetrics {
                 private NotificationsService:INotificationsService) {
     }
 
+    /**
+     * Generates metric Id given the components
+     * @param type the metric type, as a char: 'A'vailability or 'M'etric
+     * @param feedId the feed Id
+     * @param resId the resource Id
+     * @param metricId the metric Id
+     * @returns {string} a string uniquely identifying the requested metric
+     */
+    public static getMetricId(type: string, feedId: FeedId, resId: ResourceId, metricId: MetricId): string {
```

Comment:
Yeah.. but that's actual a char 'A' or 'M'.. it doesn't fit that well there. That's more of a metric type than resource type. We can certainly improve on that later